src/fosslight_binary/_jar_analysis.py

```
@@ -3,309 +3,364 @@
+            logger.debug(f"{rel_path}: Central API timed out – queued for retry (attempt 1/{_MAX_RETRY})")
+            retry_queue.append((jar_path, rel_path, sha1, 1))
+        elif result is not None:
+            jar_items[rel_path] = result
+
+    while retry_queue:
+        next_queue = []
+        for jar_path, rel_path, sha1, attempt in retry_queue:
+            if attempt >= _MAX_RETRY:
+                logger.warning(
+                    f"{rel_path}: Maven Central API timed out after {_MAX_RETRY} attempts"
+                    " – falling back to JAR internals")
```

Ruff (0.15.12)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)